assets/js/pages/Users/generatePassword.js

```diff
@@ -6,10 +6,58 @@ const SPECIAL_CHARACTERS = '~!@#$%^&*()_+={}[]|:;<>,.?/-\'"`';
 const PASSWORD_CHARACTERS =
   DIGITS + UPPERCASE_LETTERS + LOWERCASE_LETTERS + SPECIAL_CHARACTERS;
 
-export const generatePassword = (
-  length = 16,
-  characters = PASSWORD_CHARACTERS
-) =>
+export const hasValidPwdLength = (password, pwdLength = 8) =>
+  password.length >= pwdLength;
+
+export const validateNoRepetitiveCharacters = (password) =>
+  !/(.)\1{2,}/.test(password);
+
+const isAlphanumeric = (char) => /[A-Za-z0-9]/.test(char);
+
+export const validateNoSequentialCharacters = (password) => {
+  if (password.length < 3) {
+    return true;
+  }
+
+  for (let i = 2; i < password.length; i += 1) {
+    const firstChar = password[i - 2];
+    const secondChar = password[i - 1];
+    const thirdChar = password[i];
+    if (
```

Comment:
Here we exclude the special chars to ensure combinations like 89: or yz{ are still possible